launcher/modManager/cmodlistview_moc.cpp

```diff
@@ -48,6 +48,46 @@ void CModListView::changeEvent(QEvent *event)
 	QWidget::changeEvent(event);
 }
 
+void CModListView::dragEnterEvent(QDragEnterEvent* event)
+{
+	event->acceptProposedAction();
+}
+ 
+void CModListView::dragMoveEvent(QDragMoveEvent* event)
+{
+	event->acceptProposedAction();
+}
+ 
+void CModListView::dragLeaveEvent(QDragLeaveEvent* event)
+{
+	event->accept();
+}
+
+void CModListView::dropEvent(QDropEvent* event)
+{
+	const QMimeData* mimeData = event->mimeData();
+
+	if(mimeData->hasUrls())
+	{
+		QStringList pathList;
+		QList<QUrl> urlList = mimeData->urls();
+
+		for (int i = 0; i < urlList.size(); i++)
+		{
+			QString url = urlList.at(i).toString();
+			if(url.endsWith(".zip", Qt::CaseInsensitive))
+				downloadFile(url.split("/").last().toLower()
+					// mod name currently comes from zip file -> remove suffixes from github zip download
+					.replace(QRegularExpression("-[0-9a-f]{40}"), "")
+					.replace(QRegularExpression("-vcmi-.*\\.zip"), ".zip")
+					.replace(QRegularExpression("-main.zip"), ".zip")
+					, url, "mods", 0);
+			else
+				downloadFile(url.split("/").last(), url, "mods", 0);
```

Comment:
Getting last component is copy-pasted

Also, doesn't QUrl have method to get last component?